include/irr/asset/IPipelineLayout.h

```diff
@@ -10,18 +14,29 @@ namespace asset
 template<typename DescLayoutType>
 class IPipelineLayout
 {
+public:
+    struct SPushConstantRange
+    {
+        E_SHADER_STAGE stageFlags;
+        uint32_t offset;
+        uint32_t size;
+    };
+
 protected:
     virtual ~IPipelineLayout() = default;
 
-public:
     IPipelineLayout(
-        core::smart_refctd_ptr<DescLayoutType> _layout0, core::smart_refctd_ptr<DescLayoutType> _layout1 = nullptr,
+        const SPushConstantRange* const _pcRangesBegin = nullptr, const SPushConstantRange* const _pcRangesEnd = nullptr,
+        core::smart_refctd_ptr<DescLayoutType> _layout0 = nullptr, core::smart_refctd_ptr<DescLayoutType> _layout1 = nullptr,
         core::smart_refctd_ptr<DescLayoutType> _layout2 = nullptr, core::smart_refctd_ptr<DescLayoutType> _layout3 = nullptr
-    ) : m_descSetLayouts{_layout0, _layout1, _layout2, _layout3}
-    {}
+    ) : m_descSetLayouts{_layout0, _layout1, _layout2, _layout3}, 
+        m_pushConstantRanges(_pcRangesBegin==_pcRangesEnd ? nullptr : core::make_refctd_dynamic_array<core::refctd_dynamic_array<SPushConstantRange>>(_pcRangesEnd-_pcRangesBegin))
```

Comment:
isn't the syntax `core::make_refctd_dynamic_array<SPushConstantRange>` ?

Did I fuck up that function?

Reply:
nah, it's not :c

Reply:
so I fucked up ?

Reply:
apparently, if you wanted it differently. See your code with usage of it: https://github.com/Crisspl/IrrlichtBAW/blob/shaders/include/irr/video/IGPUObjectFromAssetConverter.h#L44

Reply:
I think I've fucked up slightly

would have been more sane to use`core::make_refctd_dynamic_array<core::smart_refctd_ptr<typename video::asset_traits<AssetType>::GPUObjectType> >` on that particular line and throughout the engine

Reply:
actually no I didn't fuck up, had to have it like that so `make_refctd_dynamic_array` could be used to create both refcounted and non-refcounted `dynamic_array`s

Reply:
actually no I didn't fuck up, had to have it like that so `make_refctd_dynamic_array` could be used to create both `smart_refctd_dynamic_array` and `refctd_dynamic_array`s